### DISCUSSION TITLE
How to make port accept cmake cached variable

### DISCUSSION THREAD

**Discussion by vipcxj** [microsoft/vcpkg#39241] [General]
For example, I need set CMAKE_CUDA_HOST_COMPILER to a special g++ version. Of course it works on my direct project. However it is not send to the port my project used. As a result my port always use default g++. And my cuda not accept my default g++. As a result build failed.

**Comment by Osyotr** [comment]
You're should either pass `-DCMAKE_CUDA_HOST_COMPILER=...` in `VCPKG_CMAKE_CONFIGURE_OPTIONS` triplet variable or write a CMake toolchain file that is used by both vcpkg and your project.
https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_cmake_configure_options
https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_chainload_toolchain_file
